### DISCUSSION TITLE
Entries Field: Selectable entries based on value of another field?

### DISCUSSION THREAD

**Discussion by JerikoOne** [craftcms/cms#12676] [Q&A] [answered]
Well given that WP is heading into a direction which me respectively my client doesn't need or want and were not really satisfied with our current setup anymore I'm currently assessing other options for a possible migration. So far I really like what I see here! I have one question though, please forgive me as I'm still new to Craft CMS.

Let's assume the following setup:

- Channel Locations
- Channel Rooms with an Entries Field which can reference one Entry from Channel Locations (1:n)
- Channel Events with an Entries Field, which can reference one Entry from Channel Locations (1:n)
- Channel Sessions with an Entries Field which can reference one Entry from Channel Events (1:n) and another Entries Field which can reference one Entry from Channel Rooms (1:n)

It's a basic setup to organize events, their locations and rooms - of course there are other fields, those are just the ones specific to my question. It works perfectly fine like this, however it would be great if the latter Entry Field in Channel Sessions could limit the available entries to the rooms that are actually possible (as in: referencing the correct location that is referenced by the correct event). In a relational database, I'd get the relation of possible rooms through joins (Location, then Event, where Event is the same as the one referenced in the Session), is that possible in the Crafts backend as well?

This really is just a minor issue and in no way critical, however I like my backends clean :-)

**Comment by brandonkelly** [comment, marked as answer]
This has come up a couple times in our [Discord server](https://craftcms.com/discord), and it’s technically possible to pull off with a custom condition rule, but it’s a little hacky. So I just went ahead and made it possible in core, for the upcoming Craft 4.4 release (PR’d at #12679).

So once 4.4 is out, you will be able to add a “Related To” condition rule to your “Room” field’s Selectable Entries Condition setting, set to the following template:

```twig
{event.one().location.one().id}
```

(So you’re saying that only Room entries that are related to the same location as the session’s event.)

<img width="791" alt="A Selectable Entries Condition, with a “Related To” condition rule with a dynamic element ID template" src="https://user-images.githubusercontent.com/47792/218891973-3397c995-6273-4079-9aa9-dc1e558024c6.png">

https://user-images.githubusercontent.com/47792/218893130-80c33b19-ba01-460a-bd3d-cab12451f842.mp4

**Reply by JerikoOne** [reply]
Thank you so much for the quick reply and information Brandon! Looking forward to 4.4 then!

**Reply by brandonkelly** [reply]
4.4 is out now!

**Comment by thupsi** [comment]
This is a really nice addition!
